tests/e2e/v3_curl_watch_test.go

```diff
@@ -46,3 +49,26 @@ func testCurlV3Watch(cx ctlCtx) {
 	// expects "bar", timeout after 2 seconds since stream waits forever
 	require.ErrorContains(cx.t, e2e.CURLPost(cx.epc, e2e.CURLReq{Endpoint: "/v3/watch", Value: wstr, Expected: expect.ExpectedResponse{Value: `"YmFy"`}, Timeout: 2}), "unexpected exit code")
 }
+
+// TestCurlWatchIssue19509 tries to reproduce https://github.com/etcd-io/etcd/issues/19509
+func TestCurlWatchIssue19509(t *testing.T) {
+	e2e.BeforeTest(t)
+
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+	epc, err := e2e.NewEtcdProcessCluster(ctx, t, e2e.WithConfig(e2e.NewConfigClientTLS()), e2e.WithClusterSize(1))
+	require.NoError(t, err)
+	defer epc.Close()
+
+	curlCmdAndArgs := e2e.CURLPrefixArgsCluster(epc.Cfg, epc.Procs[0], "POST", e2e.CURLReq{Endpoint: "/v3/watch", Timeout: 3})
+	for i := 0; i < 10; i++ {
```

Comment:
This can be simplified to

```suggestion
	for i := range 10 {
```

Reply:
It's not a big deal, let's just keep it as it's. I really don't want to wait another 1 hour to backport this PR, thx